### DISCUSSION TITLE
Cache counts 1 but cache list is empty

### DISCUSSION THREAD

**Discussion by muzimuzhi** [community/community#64473] [Actions]
### Select Topic Area

Bug

### Body

![image](https://github.com/community/community/assets/6376638/3671fb40-a190-4176-945d-1bb46fb5f092)
See https://github.com/muzimuzhi/latex2e/actions/caches. No workflow runs are in progress.

From a failed-to-save caching job https://github.com/muzimuzhi/latex2e/actions/runs/5921978158/job/16055705527#step:3:428, it seems the hidden cache has key `texlive-0-45f30fbcb02beecc029b93ac68c773f4bb617f95eb47d1bbcd0cd704ca8f69c6-2023-08-21`

> Failed to save: Unable to reserve cache with key texlive-0-45f30fbcb02beecc029b93ac68c773f4bb617f95eb47d1bbcd0cd704ca8f69c6-2023-08-21, another job may be creating this cache. More details: Cache already exists. Scope: refs/heads/develop, Key: texlive-0-45f30fbcb02beecc029b93ac68c773f4bb617f95eb47d1bbcd0cd704ca8f69c6-2023-08-21, Version: 9e1daa11bf29cf7f665d5c7a1aa97ff0e8b019f6083a1c921547584315ae348b
> Warning: Cache save failed.

Running ([API doc](https://docs.github.com/en/rest/actions/cache?apiVersion=2022-11-28#list-github-actions-caches-for-a-repository))
```
gh api \
  -H "Accept: application/vnd.github+json" \
  -H "X-GitHub-Api-Version: 2022-11-28" \
  /repos/muzimuzhi/latex2e/actions/caches
```
returns
```json
{
  "total_count": 1,
  "actions_caches": []
}
```

**Comment by rizul11** [comment]
See if this helps [https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows#managing-caches](url)

**Reply by muzimuzhi** [reply]
Unfortunately no.  The direct problem I encountered is that on page https://github.com/muzimuzhi/latex2e/actions/caches, it shows the total number of caches is 1, but I can't see it, hence not able to delete it.

**Comment by Jaber-Saed** [comment]
It seems like you're experiencing issues with caching in your GitHub Actions workflow. The error message you provided suggests that there might be an existing cache with the same key, and another job might be creating it concurrently. This can lead to conflicts and cache save failures.

Here are some steps you can take to address this issue:

1. **Check for Concurrent Cache Creation:**
   - The error message mentions that another job may be creating the cache concurrently. Ensure that your workflow is not creating multiple caches with the same key in parallel. If you have multiple jobs or workflows that are creating or using caches, make sure they're not conflicting with each other.

2. **Unique Cache Keys:**
   - When defining cache keys in your workflow, make sure that the keys are unique and won't collide with each other. You can include job or workflow-specific information in the cache key to differentiate between caches.

3. **Check Workflow Dependencies:**
   - If you have workflows that are dependent on the cache, ensure that they have appropriate dependencies set up. You might want to use the `needs` keyword in your workflow to ensure that jobs don't run concurrently if they share cache dependencies.

4. **Investigate Existing Caches:**
   - Check if there are any cached artifacts with the same key as mentioned in the error message. You can review your GitHub repository's "Settings" > "Actions" > "Caches" section to see if there are any existing caches that might be causing conflicts.

5. **Manual Cache Deletion:**
   - If you're sure that the cache should no longer exist and there's no job actively creating it, you might consider manually deleting the cache through the GitHub UI. This can help resolve conflicts.

6. **Review Cache Usage:**
   - Review how the caches are being used in your workflow. Ensure that the cache is only being saved and restored when it's needed. Unnecessary cache operations might lead to conflicts.

7. **Cache Management Tools:**
   - Consider using cache management tools provided by GitHub Actions, such as the `@actions/cache` action. This can help you manage cache creation and usage more effectively.

8. **GitHub Support:**
   - If the issue persists and you're unable to resolve it on your own, consider reaching out to GitHub Support. They can provide more specific assistance based on your repository's configuration and the nature of the caching issue.

It's worth noting that cache conflicts in GitHub Actions can be complex to debug, especially when multiple jobs or workflows are involved. Careful review of your workflows, cache keys, and dependencies can help identify the root cause and lead to a solution.

**Comment by pasqualesalza** [comment]
Hi, same problem here. How did you solve it? I created a ticket, it seems there’s nothing I can do.

**Comment by psychobolt** [comment]
I have same issue, as also seen in another discussion: https://github.com/orgs/community/discussions/110556 . This might be related to cache key reserves, but there is no way to delete them.

![ghost](https://github.com/community/community/assets/560721/356558e1-442d-4584-a49d-5bae617e7ae3)

gh CLI:
```
gh api -H "Accept: application/vnd.github+json"  -H "X-GitHub-Api-Version: 2022-11-28"  "/repos/{owner}/{repo}/actions/caches"
{
  "total_count": 11,
  "actions_caches": []
}
```

**Reply by muzimuzhi** [reply]
Thank you for having related tickets linked!

**Comment by DavitVosk** [comment]
I have the same problem here. I can create cache successfully, and even use it later, BUT github => caches section is empty and even showing `0` count for caches. Does anyone know what is the problem please?

<img width="1313" alt="image" src="https://github.com/user-attachments/assets/317d0b4b-edc2-41f8-a50c-1bbc2a8c6cfa" />

**Comment by teads-pierre-sabouret** [comment]
🆙  We should be able to list & delete hidden caches
